contracts/test/expanders-test.ts

```diff
@@ -507,4 +508,168 @@ describe("Expanders", async function () {
         .sub(ethers.utils.parseEther("0.1")), // Sent
     );
   });
+
+  it("should transfer ERC20 using registries and tx.origin payment", async function () {
+    const fx = await Fixture.getSingleton();
+    const { addressRegistry, blsPublicKeyRegistry } = fx.fallbackCompressor;
+    const [sendWallet, recvWallet] = await fx.createBLSWallets(2);
+    const singletonFactory = await SafeSingletonFactory.init(fx.signers[0]);
+
+    // Ballpark required payment, a bit under USD $0.01 at the time of writing
+    const txOriginPaymentAmount = ethers.utils.parseEther("0.000005");
+
+    const erc20 = await singletonFactory.connectOrDeploy(MockERC20Factory, [
+      "MockToken",
+      "MCK",
+      0,
+    ]);
+
+    await Promise.all([
+      addressRegistry.register(erc20.address),
+      receiptOf(
+        fx.signers[0].sendTransaction({
+          to: sendWallet.address,
+          value: txOriginPaymentAmount,
+        }),
+      ),
+      ...[sendWallet, recvWallet]
+        .map((wallet) => [
+          blsPublicKeyRegistry.register(wallet.PublicKey()),
+          addressRegistry.register(wallet.address),
+          receiptOf(erc20.mint(wallet.address, ethers.utils.parseEther("1.0"))),
+        ])
+        .flat(),
+    ]);
+
+    const bundle = await sendWallet.signWithGasEstimate({
+      nonce: await sendWallet.Nonce(),
+      actions: [
+        {
+          ethValue: 0,
+          contractAddress: erc20.address,
+          encodedFunction: erc20.interface.encodeFunctionData("transfer", [
+            recvWallet.address,
+            ethers.utils.parseEther("0.1"),
+          ]),
+        },
+        {
+          ethValue: txOriginPaymentAmount,
+          contractAddress: fx.utilities.address,
+          encodedFunction:
+            fx.utilities.interface.encodeFunctionData("sendEthToTxOrigin"),
+        },
+      ],
+    });
+
+    /*
+      Example:
+
+      01      - One operation
+      02      - Use expander 2 (ERC20 expander)
+      0f      - 0x0f = 0b1111 bit stream:
+                - 1: Use registry for BLS key
+                - 1: Include a tx.origin payment
+                - 1: Use registry for ERC20 address
+                - 1: Use registry for recipient address
+      000000  - Registry index for sendWallet's public key
+      00      - nonce: 0
```

Comment:
Nah, we're using VLQ for the nonce, which stands for Variable Length Quantity. This means the encoding expands as needed to accommodate large values - this will work for the full `uint256` range (and theoretically beyond!).

Examples:

```
 * 0x00: 0
 * 0x01: 1
 * 0x7f: 127
 * 0x8100: 128
 * 0x8105: 133 = 1 * 128 + 5
 * 0x838005: 49,157 = 3 * 128^2 + 0 * 128^1 + 5 * 128^0
```

More information: https://github.com/web3well/bls-wallet/blob/8c91d2f/contracts/contracts/lib/VLQ.sol